### PR TITLE
SISRP-40607 - Summer Late petition eform link not showing up UNODG students

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-40607

https://jira.berkeley.edu/browse/SISRP-39160

See QA PR #7195

### CODE REVIEW

File: src/assets/javascripts/angular/controllers/widgets/academics/graduateDegreeProgressController.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ angular.module('calcentral.controllers').controller('GraduateDegreeProgressContr
       $scope.showEmptyMilestonesMessage = apiService.user.profile.academicRoles.current.grad || (apiService.user.profile.academicRoles.current.law && apiService.user.profile.academicRoles.current.lawJspJsd);
       var hasAprLink = !!$scope.degreeProgress.graduate.links;
       var hasMilestones = $scope.degreeProgress.graduate.progresses.length;
-      $scope.degreeProgress.graduate.showCard = apiService.user.profile.features.csDegreeProgressGradStudent && (hasAprLink || hasMilestones || $scope.showEmptyMilestonesMessage) && !apiService.user.profile.academicRoles.current.nonDegreeSeekingSummerVisitor;
+      $scope.degreeProgress.graduate.showCard = apiService.user.profile.features.csDegreeProgressGradStudent && (hasAprLink || hasMilestones || $scope.showEmptyMilestonesMessage) && !academicsService.isNonDegreeSeekingSummerVisitor(apiService.user.profile.academicRoles);
```
Comment (davidlimcheng) [inline src/assets/javascripts/angular/controllers/widgets/academics/graduateDegreeProgressController.js]: @redconfetti, I see that the old behavior was looking at the current roles, while this change looks at the historical roles:
```
  var isNonDegreeSeekingSummerVisitor = function(academicRoles) {
    return academicRoles.historical.summerVisitor && !academicRoles.historical.degreeSeeking;
  };
```
Will this lead to any unexpected behavior?

**Comment by davidlimcheng** [comment]
Looks good @redconfetti , just one question for you.  Also, can you explain a bit how this relates to [SISRP-39160](https://jira.berkeley.edu/browse/SISRP-39160)?

File: src/assets/javascripts/angular/controllers/widgets/academics/graduateDegreeProgressController.js (hunk 1 above)
Comment (redconfetti) [reply src/assets/javascripts/angular/controllers/widgets/academics/graduateDegreeProgressController.js]: @davidlimcheng - No. For that functionality, it is expected that it be based on historical roles, not current.

Comment (redconfetti) [reply src/assets/javascripts/angular/controllers/widgets/academics/graduateDegreeProgressController.js]: @davidlimcheng - This is the line that is related to SISRP-39160. I line item'ed it, like a congress man.

**Pull request merged**